### PR TITLE
Present a more 12 factor app style config.py file

### PR DESCRIPTION
**Pull request by carsongee**
Show how to set certificates and other secrets using environment variables per http://12factor.net/

### CODE REVIEW

File: config.py [hunk 1]
```diff
@@ -1,12 +1,35 @@
...
+# ```
+# export CONSUMER_KEY_CERT=$(cat <<EOF
+# < paste cert here>
+# EOF
+# )
+# ```
+
+CONSUMER_KEY_PEM_FILE = os.path.abspath('consumer_key.pem')
+with open(CONSUMER_KEY_PEM_FILE, 'w') as wfile:
+    wfile.write(os.environ.get('CONSUMER_KEY_CERT', ''))
```
Comment (pwilkins) [inline config.py]: The cert text is entered in the config.py file and then on start-up, that text is written to a file on disk?  If you are already editing a file to store the cert text, why programmatically write the cert text to disk?  What am I overlooking?

File: docs/deploy_to_heroku.rst [hunk 2]
```diff
@@ -88,6 +88,65 @@ The ``create`` command returns useful information.  It indicates
...
+OAuth keys needed for LTI security.  As a result, we have instrumented
+the ``config.py`` file to have an example of using environment
+variables to store these secrets.
+
+For example:
+
+.. code-block:: python
+
+  # SECRET KEY FOR AUTHENTICATION
+  SECRET_KEY = OS.ENVIRON.GET("FLASK_SECRET_KEY", "YOU-WILL-NEVER-GUESS")
```
Comment (pwilkins) [inline docs/deploy_to_heroku.rst]: shouldn't OS.ENVIRON.GET be lower case?

Comment (carsongee) [inline docs/deploy_to_heroku.rst]: yeah, emacs upcase typo here.  I've resolved that

File: config.py (hunk 1 above)
Comment (carsongee) [inline config.py]: the lack of security of a github repo.  The SSL library requires a file in order to do certificate authentication, but heroku only allows you to add files via a git repo or store them somewhere else like S3.  So we just do the conversion on startup from environment to file

**Pull request merged**